crates/rattler_solve/src/resolvo/mod.rs

```diff
@@ -651,16 +747,33 @@ impl super::SolverImpl for Solver {
                 .intern_version_set(name_id, NamelessMatchSpec::default().into())
         });
 
-        let root_requirements = task.specs.iter().map(|spec| {
+        let root_requirements = task.specs.iter().flat_map(|spec| {
             let (name, nameless_spec) = spec.clone().into_nameless();
+            let features = spec.optional_features.clone();
             let name = name.expect("cannot use matchspec without a name");
             let name_id = provider.pool.intern_package_name(name.as_normalized());
-            provider
+            let mut reqs = vec![provider
                 .pool
-                .intern_version_set(name_id, nameless_spec.into())
+                .intern_version_set(name_id, nameless_spec.clone().into())];
+
+            // Add requirements for optional features if specified
+            if let Some(features) = features {
+                for feature in features {
+                    // Create a version set that matches the feature-enabled package
+                    let package_name_with_feature =
+                        format!("{}[{}]", name.as_normalized(), feature);
+                    let feature_name_id =
+                        provider.pool.intern_package_name(package_name_with_feature);
+                    let feature_version_set = provider
+                        .pool
+                        .intern_version_set(feature_name_id, nameless_spec.clone().into());
+                    reqs.push(feature_version_set);
```

Comment:
You could also just clone the `spec` and add the optional dependency perhaps? Instead of parsing.